src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpTestForEmptyStringsUsingStringLength.Fixer.cs

```diff
@@ -37,5 +37,21 @@ protected override SyntaxNode GetRightOperand(SyntaxNode binaryExpressionSyntax)
         {
             return ((BinaryExpressionSyntax)binaryExpressionSyntax).Right;
         }
+
+        protected override bool IsFixableBinaryExpression(SyntaxNode node)
+        {
+            return (node is BinaryExpressionSyntax) && (IsEqualsOperator(node) || IsNotEqualsOperator(node));
+        }
+
+        protected override bool IsFixableInvocationExpression(SyntaxNode node)
+        {
+            return node.IsKind(SyntaxKind.InvocationExpression);
+        }
+
+        protected override SyntaxNode GetInvocationTarget(SyntaxNode node)
+        {
+            MemberAccessExpressionSyntax expression = (MemberAccessExpressionSyntax)((InvocationExpressionSyntax)node).Expression;
```

Comment:
It feels quite unsafe to use direct cast.

Reply:
Would it be preferable to check the type first and attempt to exit early and not emit a code fix in case there is a type mismatch?